metadata.json

```diff
@@ -102,7 +102,7 @@
       "version_requirement": ">= 6.0.0 < 8.0.0"
     }
   ],
-  "pdk-version": "2.3.0",
-  "template-url": "pdk-default#2.3.0",
-  "template-ref": "tags/2.3.0-0-g8aaceff"
+  "pdk-version": "2.5.0",
```

Comment:
There is really a lot going on in this PR. Could you submit the pdk update in a dedicated PR? that makes it easier to review and gives a better changelog. The current PR title doesn't really reflect all the changes.

Reply:
Fresh PR [here](https://github.com/puppetlabs/puppetlabs-patching_as_code/pull/81), I can rebase this PR after that one has been merged

Reply:
could we get some of the commits squashed too?